### PR TITLE
Grouping and mapping extraction client.

### PR DESCRIPTION
**Pull request by julioalvahash**
New grouping and mapping extraction client.
- Removed old extraction client from reporting, updated imports.
- Removed old interfaces.
- New integration and unit test for new extraction client. 
- Removed old extraction client functions from validation test.
- Updated Odata integration test to use new extraction client.

### CODE REVIEW

File: src/grouping-and-mapping/clients/ExtractionClient.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        "Parameter top was outside of the valid range [1-1000]."
+      );
+    }
+    const url = `${this._baseUrl}?iModelId=${iModelId}${top ? `&$top=${top}` : `` }`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    const response = await this.fetchJSON<ExtractionsResponse>(url, requestOptions);
+    return response;
+  }
+
+  public getIModelExtractionsIterator(accessToken: AccessToken, iModelId: string, top?: number | undefined): EntityListIterator<ExtractionStatus> {
```
Comment (Guillar1) [inline src/grouping-and-mapping/clients/ExtractionClient.ts:41]: `undefined` not needed.

File: src/grouping-and-mapping/interfaces/IExtractionClient.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+   */
+  getIModelExtractionsIterator(accessToken: AccessToken, iModelId: string, top?: number): EntityListIterator<ExtractionStatus>;
+
+  /**
+   * Gets the logs of an extraction.
+   * @param {AccessToken} accessToken OAuth access token with imodels:read or itwin-platform scope.
+   * @param {string}extractionId The extraction Id.
+   * @param {number} top Optional max items to be sent in response.
+   * @memberof ExtractionClient
+   * @link "https://developer.bentley.com/apis/grouping-and-mapping/operations/get-extraction-logs/"
```
Comment (Guillar1) [inline src/grouping-and-mapping/interfaces/IExtractionClient.ts:54]: Quotation marks around the link. This pattern doesn't match.

File: src/grouping-and-mapping/interfaces/IExtractionClient.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+   * @param {number} top Optional max items to be sent in response.
+   * @memberof ExtractionClient
+   * @link https://developer.bentley.com/apis/grouping-and-mapping/operations/get-imodel-extractions/
+   */
+  getIModelExtractionsIterator(accessToken: AccessToken, iModelId: string, top?: number): EntityListIterator<ExtractionStatus>;
+
+  /**
+   * Gets the logs of an extraction.
+   * @param {AccessToken} accessToken OAuth access token with imodels:read or itwin-platform scope.
+   * @param {string}extractionId The extraction Id.
```
Comment (Guillar1) [inline src/grouping-and-mapping/interfaces/IExtractionClient.ts:51]: Missing space between `{string}` and `extractionId`

File: src/grouping-and-mapping/clients/ExtractionClient.ts (hunk 1 above)
Comment (julioalvahash) [reply src/grouping-and-mapping/clients/ExtractionClient.ts]: It get auto generated by vscode for some reason, I removed them.

File: src/grouping-and-mapping/interfaces/IExtractionClient.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+   * @param {string} extractionId The extraction Id.
+   * @memberof ExtractionClient
+   * @link https://developer.bentley.com/apis/grouping-and-mapping/operations/get-extraction-status/
+   */
+  getExtractionStatus(accessToken: AccessToken, extractionId: string): Promise<ExtractionStatus>;
+
+  /**
+   * Gets all of the extractions that were run on a given iModel.
+   * @param {AccessToken} accessToken OAuth access token with imodels:read or itwin-platform scope.
+   * @param {string} iModelId The extraction Id.
```
[the remark this answers is not in the document]
Comment (julioalvahash) [reply src/grouping-and-mapping/interfaces/IExtractionClient.ts]: Changed to `The iModel id`

**Review by Guillar1 — APPROVED** [review]

**Pull request merged**